fvm/evm/emulator/emulator.go

```diff
@@ -234,7 +234,17 @@ func (bl *BlockView) DryRunTransaction(
 	msg.SkipAccountChecks = true
 
 	// return without commiting the state
-	return proc.run(msg, tx.Hash(), 0, tx.Type())
+	txResult, err := proc.run(msg, tx.Hash(), 0, tx.Type())
+	if txResult.VMError == nil && txResult.ValidationError == nil {
+		// Adding `gethParams.SstoreSentryGasEIP2200` is needed for this condition:
+		// https://github.com/onflow/go-ethereum/blob/master/core/vm/operations_acl.go#L29-L32
+		txResult.GasConsumed += gethParams.SstoreSentryGasEIP2200
+		// Take into account any gas refunds, which are calculated only after
+		// transaction execution.
+		txResult.GasConsumed += txResult.GasRefund
```

Comment:
wouldn't this be relevant also for other calls not just dry run?

Reply:
Because `EVM.dryRun` does not commit any changes, there is no gas refunds after its successful execution, so we must include it in the `GasConsumed` field.
In other calls, all state changes are committed. This means that any gas refunds will be returned back to the TX author, and the `GasConsumed` field should reflect the actual gas consumed, without including the gas refunds.